### PR TITLE
feat(data-grid) Add actions cell type

### PR DESCRIPTION
**Pull request by sagi-T**

### CODE REVIEW

**Review by acstll — CHANGES_REQUESTED** [review]
Thank you @sagi-T this is great!

My only comment is regarding a little change in the CSS.

And also, I would ask you to run the code formatter too (you can do `yarn format` from the root folder or from `packages/components`).

Before this is ready to merge, we need to update the documentation (Storybook). There's some info regarding cell types in `packages/storybook-vue/stories/3_components/data-grid/DataGrid.stories.mdx`…

Is this something you'd like to give it a try? I can also help, just let me know.

Thanks again!

File: packages/components/src/components/data-grid/data-grid.css [hunk 1]
```diff
@@ -384,6 +384,9 @@
 .tbody__cell scale-progress-bar {
   flex-grow: 1;
 }
+.tbody__actions scale-button {
+  margin-right: 0.5em;
```
[the remark this answers is not in the document]
Comment (sagi-T) [reply packages/components/src/components/data-grid/data-grid.css]: thanks, it is fixed! :)

**Comment by acstll** [comment]
@sagi-T thanks again for the update!

We were discussing yesterday that it would be nice to offer the possibility also to have buttons with icons or icon-only buttons.

I thought about this for the implementation:

```diff
const { label, ...props } = action;
+ if (typeof label === 'object' && '__html' in label) {
+   return <scale-button innerHTML={label.__html} {...props}></scale-button>;
+ }
return <scale-button {...props}>{label}</scale-button>;
```

so one can replace the `label` with some HTML, but we still keep `label` as the only prop in the row object not going into the button as attribute, which I think is important to keep the API clean and consistent.

```js
{
  label: {
    __html: `<scale-icon-action-edit></scale-icon-action-edit>`
  },
  variant: 'secondary',
  iconOnly: 'true',
  ariaLabel: 'Edit',
  onClick: () => { /* ... */ },
}
```

Do you agree? would you do it differently?

**Comment by sagi-T** [comment]
Hi @acstll. No I don't, I totally agree with you because these options are very important in this concept and also for us, We just didn't think of it :D So thank you. I will make the change as soon as possible.

if it is okay for you. :)

**Comment by acstll** [comment]
> if it is okay for you. :)

totally OK 👍 

I will then take care of the documentation part afterwards.

Thanks again!

**Comment by sagi-T** [comment]
Thank you @acstll
I'm Finished. We are looking forward to the documentation part from your side :D 
After is it merged when can we expect this to be released? We would like to use this component as soon as possible 💯

**Review by acstll — APPROVED** [review]

**Comment by acstll** [comment]
thank you @sagi-T! 

I'll take care of the docs in a different PR.

We can make a release tomorrow, with the feature undocumented, so you can start using it right away :)

The prettier check is still failing (which is weird, since you already did that in https://github.com/telekom/scale/pull/444/commits/78ec7b86886f664d266c34f4483060c761a399c8); could you please try and run `yarn format` once more inside packages/components/?

image

**Comment by sagi-T** [comment]
> thank you @sagi-T!
> 
> I'll take care of the docs in a different PR.
> 
> We can make a release tomorrow, with the feature undocumented, so you can start using it right away :)
> 
> The prettier check is still failing (which is weird, since you already did that in [78ec7b8](https://github.com/telekom/scale/commit/78ec7b86886f664d266c34f4483060c761a399c8)); could you please try and run `yarn format` once more inside packages/components/?
> 
> image

Yes of course :)

**Comment by sagi-T** [comment]
> > thank you @sagi-T!
> > I'll take care of the docs in a different PR.
> > We can make a release tomorrow, with the feature undocumented, so you can start using it right away :)
> > The prettier check is still failing (which is weird, since you already did that in [78ec7b8](https://github.com/telekom/scale/commit/78ec7b86886f664d266c34f4483060c761a399c8)); could you please try and run `yarn format` once more inside packages/components/?
> > image
> 
> Yes of course :)

Hmm🧐. Interesting I did that a second ago and everything is fine nothing changes on the files.
Am I do something  wrong? :D
image

**Comment by acstll** [comment]
> Hmm🧐. Interesting I did that a second ago and everything is fine nothing changes on the files.
> Am I do something wrong? :D

that is weird indeed, can you try with yarn instead of npm? (we use Yarn v1) — that makes a difference sometimes 🤷 
also, you can try `yarn format` directly from the `packages/components` directory, not from the root.

Let me know how that goes.

(I can still merge it and fix it later.)

Thank you 🙏

**Comment by sagi-T** [comment]
> > Hmm🧐. Interesting I did that a second ago and everything is fine nothing changes on the files.
> > Am I do something wrong? :D
> 
> that is weird indeed, can you try with yarn instead of npm? (we use Yarn v1) — that makes a difference sometimes 🤷
> also, you can try `yarn format` directly from the `packages/components` directory, not from the root.
> 
> Let me know how that goes.
> 
> (I can still merge it and fix it later.)
> 
> Thank you 🙏

Yay 🥶 So now I used yarn format instead of npm in the `packages/components` directory  as you can see the picture below 👇 

image

But nothing has changed 😮‍💨 Sorry for that @acstll. Any idea  for the possible solution ? 
I'm looking forward to your response and thank you for your patience 🙏

**Comment by acstll** [comment]
No worries, I'm gonna merge and then re-check if the prettier issue persist, and fix it in another PR :)

**Comment by acstll** [comment]
Thank you very much, once more @sagi-T 

We'll make the release on Monday 🙏

**Pull request merged**